File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
// Prompts user for valid numerical key
    int key;
    if(argc == 2)
    {
        for(int i = 0, n = strlen(argv[1]); i < n; i++)
        {
            if(!(isdigit(argv[1][i])))
            {
                printf("Usage: ./caesar key\n");
                return 1;
            }
        }
        key = atoi(argv[1]);
    }
    else
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
// Prompts user for plain text
    string p = get_string("plaintext: ");
// Encrypt the text and displays cipher text
    printf("ciphertext: ");
    for(int i = 0, n = strlen(p); i < n; i++)
    {
        if(islower(p[i]))
        {
            printf("%c", 'a'+ (p[i] + key - 'a') % 26);
        }
        else if(isupper(p[i]))
        {
            printf("%c", 'A' + (p[i] + key - 'A') % 26);
        }
        else
        {
            printf("%c", p[i]);
        }
    }
    printf("\n");
    return 0;
}

File: pset6/caesar/caesar.py
from sys import argv
from sys import exit
from cs50 import get_string

if len(argv) != 2:
    print("usage: python caesar.py key")
    exit(1)

key = int(argv[1]) % 26
raw = get_string("plaintext: ")

# Print encrypted text
print("ciphertext: ", end="")
for c in raw:
    if c.islower():
        print(f"{chr(ord('a') + (ord(c) + key - ord('a')) % 26)}", end="")
    elif c.isupper():
        print(f"{chr(ord('A') + (ord(c) + key - ord('A')) % 26)}", end="")
    else:
        print(f"{c}", end="")
print("")